tests/unit/test_charm.py

```diff
@@ -254,151 +253,64 @@ def test_iscsi_with_initiatorname_rendered(mocker, harness, iscsi_config):
+):
+    """Test iscsi config when hostname is absent from initiator-dictionary."""
     mocker.patch("charm.utils.is_container", return_value=False)
-    mocker.patch("charm.subprocess.getoutput", return_value="iqn.2020-07.canonical.com:lun1")
+    mock_getoutput = mocker.patch(
+        "charm.subprocess.getoutput", return_value="iqn.2020-07.canonical.com:lun1"
+    )
     mocker.patch("charm.socket.getfqdn", return_value="testhost.testdomain")
     mocker.patch("charm.subprocess.check_call")
     mocker.patch("charm.subprocess.check_output")
     mocker.patch("charm.StorageConnectorCharm._configure_deferred_restarts")
     mocker.patch("charm.StorageConnectorCharm._defer_service_restart")
     mocker.patch("charm.Path.chmod")
-
-    # test scenario 1
-    # initiator-dictionary configured with same name as the one present in initiatorname.iscsi
-    expected_initiator_content = dedent(
-        """\
-        ###############################################################################
-        # [ WARNING ]
-        # configuration file maintained by Juju
-        # local changes will be overwritten.
-        #
-        # DO NOT EDIT OR REMOVE THIS FILE!
-        # If you remove this file, the iSCSI daemon will not start.
-        # If you change the InitiatorName, existing access control lists
-        # may reject this initiator.  The InitiatorName must be unique
-        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
-        ###############################################################################
-        InitiatorName=iqn.2020-07.canonical.com:lun1"""
+    mock_getoutput = mocker.patch(
+        "charm.subprocess.getoutput", return_value="iqn.2020-07.canonical.com:lun1"
     )
-    initiator_content_with_same_name = expected_initiator_content
-    harness.charm.ISCSI_INITIATOR_NAME.write_text(initiator_content_with_same_name)
-    mock_write_text = mocker.patch("charm.Path.write_text")
-
-    harness.charm._stored.installed = True
-    harness.update_config(iscsi_config)
-
-    assert harness.charm._get_initiator_name_from_file() == "iqn.2020-07.canonical.com:lun1"
-    # initiatorname.iscsi not rendered again
-    # assert that there are no Path.write_text calls from _iscsi_initiator.
-    # only write_text calls present are the ones from _multipath_configuration and
-    # _iscsid_configuration
-    assert len(mock_write_text.mock_calls) == 2
-    assert isinstance(harness.charm.unit.status, ActiveStatus)
+    mock_render = mocker.patch("charm.StorageConnectorCharm._render_iscsi_initiator")
 
-    mock_write_text.reset_mock()
+    if initiator_content is not None:
+        harness.charm.ISCSI_INITIATOR_NAME.write_text(initiator_content)
 
-    # test scenario 2
-    # initiator name present in initiatorname.iscsi and nothing configured in initiator-dictionary
-    # reuse the same name and don't render the file again
     iscsi_config["initiator-dictionary"] = "{}"
```

Comment:
Could we add an additional happy test case to test different iscsi_config["initiator-dictionary"] ?
a. hostname is available
b. hostname is not available

Reply:
We're already testing both cases of `iscsi_config["initiator-dictionary"]`
1. hostname is available: This is the default case which is being tested in `test_iscsi_with_initiator_dictionary_hostname`. The `iscsi_config` fixture (present in `conftest.py`) sets up `initiator-dictionary` with the hostname.
2. hostname is not available: This is the case you've highlighted in your comment. I've explicitly configured `initiator-dictionary` with no hostname.

Or maybe did I not understand you correctly?

Reply:
What if hostname is not available in the dictonary (but it's not None)? 
So far, I think the code is good enough, we can skip it this time.